app/scripts/metamask-controller.js

```diff
@@ -4788,6 +4788,24 @@ export default class MetamaskController extends EventEmitter {
     } ${hdPathDescription || ''}`.trim();
   }
 
+  /**
+   * Set account name that checks for identical label
+   *
+   * @param account
+   * @param index
+   * @param hwDeviceName
```

Comment:
Nit: can we please call this `hardwareDeviceName` rather than `hwDeviceName`? As a general rule we try to avoid contractions, as they can be difficult to read (especially for contributors not that familiar with this area).